avocado/core/runners/avocado_instrumented.py

```diff
@@ -104,12 +104,20 @@ def _run_avocado(runnable, queue):
             module_path, klass_method = runnable.uri.split(':', 1)
 
             klass, method = klass_method.split('.', 1)
+
+            # inject the run.test_parameters option in the Test params
+            # this handles the -p command-line argument
+            test_parameters = {name: value for name, value in
+                               runnable.config.get('run.test_parameters', [])}
+            params = TreeNode().get_node("/", True)
+            params.value = test_parameters
+
```

Comment:
IIUC, `run.test_parameters` is a list of lists. Something like:

```python
"run.test_parameters": [["foo", "1"]]
```

So what do you think about this:

```python
params = TreeNode().get_node("/", True)
params.value = dict(runnable.config.get('run.test_parameters'))
```

Note: AFAICT, config.get() should return all the values, and the default value for `run.test_parameters` is already a empty list. So, when getting, the default `[]` wouldn't be necessary. But if that is not the case here, then you can keep the `[]`.

Reply:
Much better, thanks! I force-pushed the change.